### ISSUE TITLE
v2.0.0 CPU usage and unable to type in terminal

### ISSUE THREAD

**Issue by logix2** [GitSquared/edex-ui#388]
On Linux, the latest eDEX-UI v2.0.0 spawns some processes that use a lot of CPU, like "cat /dev/urandom", "tr -dc -azA-Z0-9" and "fold -w 24".  Also, the terminal never gets past "Welcome to eDEX-UI v2.0.0 - Electron v4.0.1" so I can't type anything in the eDEX-UI terminal.

Edit -> log:

```
▶  start     Starting eDEX-UI v2.0.0
ℹ  info      With Node 10.11.0 and Electron 4.0.1
ℹ  info      Renderer is Chrome 69.0.3497.106
▶  Startup   Initialized timer...
ℹ  info      Base config dir is /home/logix/.config/eDEX-UI
☐  pending   Loading settings file...
✔  success   Settings loaded!
☐  pending   Creating new terminal process on port 3000
✔  success   Terminal back-end initialized!
☐  pending   Starting multithreaded calls controller...
✔  success   Multithreaded controller ready
ℹ  info      Creating window...
ℹ  info      Multithread worker started at 21342
…  watching  Waiting for frontend connection...
ℹ  info      Multithread worker started at 21343
ℹ  info      Multithread worker started at 21345
ℹ  info      Multithread worker started at 21347
ℹ  info      Multithread worker started at 21348
ℹ  info      Multithread worker started at 21349
ℹ  info      Multithread worker started at 21350
/sys/firmware/dmi/tables/smbios_entry_point: Permission denied
/dev/mem: Permission denied
✔  success   Connected to frontend!
◼  Startup   Timer run for: 23.79s
ℹ  info      Resized TTY to  138 030
ℹ  info      UpdateChecker: Running latest version.
ℹ  info      All windows closed
☒  complete  Shutting down...
☒  complete  Terminal exited 0 1
```

I guess it wants to run as root? That's not a good idea...

Edit 2: Using Ubuntu 18.10 (64bit).

**Comment by GitSquared** [comment]
Wow, what the hell? Did v1.1.2 work on your setup? It looks like `systeminformation` fails to access memory information, but that should normally be available without root privileges.

**Comment by logix2** [comment]
Previous versions now seem to have the same issue somehow, but at least one version previously worked - I even wrote an article about it. Here's the 1.1.2 log:

```
▶  start     Starting eDEX-UI v1.1.2
ℹ  info      With Node 8.9.3 and Electron 2.0.14
ℹ  info      Renderer is Chrome 61.0.3163.100
▶  Startup   Initialized timer...
ℹ  info      Base config dir is /home/logix/.config/eDEX-UI
☐  pending   Loading settings file...
✔  success   Settings loaded!
☐  pending   Creating new terminal process on port 3000
✔  success   Terminal back-end initialized!
ℹ  info      Creating window...
…  watching  Waiting for frontend connection...
✔  success   Connected to frontend!
◼  Startup   Timer run for: 22.64s
ℹ  info      Resized TTY to  089 038
ℹ  info      Resized TTY to  138 035
```
I tried deleting the configuration directory but without any success.

**Comment by GitSquared** [comment]
I'm afraid this might be an issue with your setup, but it seems really weird. Obviously i can't reproduce this so far.

Do you have node+npm installed? We might test some things directly if you do.

**Comment by erixom** [comment]
Oh gotcha, my bad prob shoulda checked that.  Do you still have an open
issue for the issue on Windows?

**Comment by logix2** [comment]
It doesn't show any permission issues  on older eDEX-UI but it still doesn't work. So it's probably something else. By the way, I'm seeing the update notifier for the first time with the 2.0.0 release, maybe that's somehow triggering this?

Yes, I have node+npm installed and setup.

**Comment by GitSquared** [comment]
@logix2 What do you mean by "seeing the update notifier"? As in, in the logs or do you get the graphical prompt in eDEX that a new update has been released? The update notifier has been here since v1.0 I believe.

Could please clone the repo and run eDEX from source (`npm run install-linux` then `npm start`) to see if the issue still appears?

If it still does, I might try to send you some builds of eDEX with specific modules disabled to try and pinpoint the problem. Otherwise, the issue is with AppImage packaging...

Thanks for your help.

**Comment by logix2** [comment]
Yes, I meant the prompt that lets users know about a new eDEX version, I don't remember seeing it before.

I ran ``npm run install-linux && npm start`` and the same issue occurs: cat, tr and fold processes using a lot of CPU and I can't type in eDEX (the terminal prompt never shows, it gets stuck to "Welcome to eDEX-UI ...".

**Comment by GitSquared** [comment]
@logix2 I found something that may be related to this issue while browsing the latest changes in [systeminformation](https://github.com/sebhildebrandt/systeminformation): there's [this issue](https://github.com/sebhildebrandt/systeminformation/issues/193) that mentions the same error message you got and was fixed in the latest (4.0) release of SI less than 24 hours ago.

The master branch has been [updated](https://github.com/GitSquared/edex-ui/commit/eea54570b84108ace90cf019d8386bd8fd955052) to include this, could you try and run eDEX from source again?

**Comment by GitSquared** [comment]
Closed as I believe the issue to be fixed.
